src/vellum/workflows/resolvers/resolver.py

```diff
@@ -16,27 +18,66 @@ def get_latest_execution_events(self) -> Iterator[WorkflowEvent]:
     def get_state_snapshot_history(self) -> Iterator[BaseState]:
         return iter([])
 
-    def load_state(self, previous_execution_id: Optional[UUID] = None) -> Optional[BaseState]:
+    def _find_execution_events(
+        self, execution_id: str, spans: list
+    ) -> Tuple[Optional[WorkflowExecutionInitiatedEvent], Optional[WorkflowExecutionInitiatedEvent]]:
+        previous_invocation: Optional[WorkflowExecutionInitiatedEvent] = None
+        root_invocation: Optional[WorkflowExecutionInitiatedEvent] = None
+
+        for span in spans:
+            # Look for workflow execution spans
+            if isinstance(span, WorkflowExecutionSpan):
+                # Find the WorkflowExecutionInitiatedEvent in the span's events
+                for event in span.events:
+                    if isinstance(event, WorkflowExecutionInitiatedEvent):
+                        # Check if this is the previous execution matches the execution_id
+                        if event.span_id == execution_id:
+                            previous_invocation = event
+
+                        if not event.links:
+                            root_invocation = event
```

Comment:
[blocking] this is not currently correct, since this is treating every span as a possible root invocation. Once we find the previous span, we check _its_ links and see it's root invocation to reuse